heron/scheduler-core/src/java/com/twitter/heron/scheduler/dryrun/FormatterUtils.java

```diff
@@ -0,0 +1,470 @@
+     * Format rows and title into a table
+     * @return Formatted table
+     */
+    public String createTable() {
+      // Generate formatter for each cell in a single row
+      List<String> formatters = generateRowFormatter();
+      // Set formatter for each row
+      title.setFormatters(formatters);
+      for (Row row: rows) {
+        row.setFormatters(formatters);
+      }
+      // Calculate length for frames
+      int frameLength = caculateFrameLength();
+      // Start building table
+      StringBuilder builder = new StringBuilder();
+      // Add upper frame
+      addRow(builder, Strings.repeat("=", frameLength));
+      // Add title
+      addRow(builder, title.toString());
+      // Add one single line to separate title and content
+      addRow(builder, Strings.repeat("-", frameLength));
+      // Add each row
+      for (Row row: rows) {
+        addRow(builder, row.toString());
+      }
+      // Add lower frame
+      addRow(builder, Strings.repeat("=", frameLength));
+      return builder.toString();
+    }
+  }
+
+
+  /******************************** Auxiliary functions ********************************/
+
+  /**
+   * Format new amount associated with change in percentage
+   * For example, with {@code oldAmount = 2} and {@code newAmount = 1}
+   * the result Cell is " -50.00%" (in red color)
+   * @param oldAmount old resource usage
+   * @param newAmount new resource usage
+   * @return formatted chagne in percentage if oldAmount and newAmount differ
+   */
+  public static Optional<Cell> percentageChange(double oldAmount, double newAmount) {
+    double delta = newAmount - oldAmount;
+    double percentage = delta / oldAmount * 100.0;
+    if (percentage == 0.0) {
+      return Optional.absent();
+    } else {
+      String sign = "";
+      if (percentage > 0.0) {
+        sign = "+";
+      }
+      Cell cell = new Cell(String.format("%s%.2f%%", sign, percentage));
+      // set color to red if percentage drops, to green if percentage increases
+      if ("".equals(sign)) {
+        cell.setColor(TextColor.RED);
+      } else {
+        cell.setColor(TextColor.GREEN);
+      }
+      return Optional.of(cell);
+    }
+  }
+
+  public static Row rowOfInstancePlan(PackingPlan.InstancePlan plan,
+                                      TextColor color, TextStyle style) {
+    String taskId = String.valueOf(plan.getTaskId());
+    String cpu = String.valueOf(plan.getResource().getCpu());
+    String ram = String.valueOf(plan.getResource().getRam().asGigabytes());
+    String disk = String.valueOf(plan.getResource().getDisk().asGigabytes());
+    List<String> cells = Arrays.asList(
+          plan.getComponentName(), taskId, cpu, ram, disk);
+    Row row = new Row(cells);
+    row.setStyle(style);
+    row.setColor(color);
+    return row;
+  }
+
+  public static String renderOneContainer(List<Row> rows) {
+    List<String> titleNames = Arrays.asList(
+        "component", "task ID", "CPU", "RAM (GB)", "disk (GB)");
```

Comment:
A tiny comment. I was wondering if the `titleNames` variable could be a static and immutable `List` instead of being create every times when the function invoked since it was never changed in every single table.